### PR TITLE
Basic tests for transcoding

### PR DESCRIPTION
**Pull request by lizan**

### CODE REVIEW

File: src/envoy/transcoding/config.cc [hunk 1]
```diff
@@ -135,8 +135,9 @@ Config::Config(const Json::Object& config, Server::Instance& server) {
           break;
       }
 
-      pmb.Register("POST", "/" + service->full_name() + "/" + method->name(),
-                   "", method_info);
+      //      pmb.Register("POST", "/" + service->full_name() + "/" +
```
Comment (qiwzhang) [inline src/envoy/transcoding/config.cc]: why do we comment this out?

Comment (lizan) [inline src/envoy/transcoding/config.cc]: This was a debug registration before supporting path binding. Removed.

File: src/envoy/transcoding/config.cc [hunk 2]
```diff
@@ -134,10 +134,6 @@ Config::Config(const Json::Object& config) {
         default:
           break;
       }
-
-      //      pmb.Register("POST", "/" + service->full_name() + "/" +
```
Comment (qiwzhang) [inline src/envoy/transcoding/config.cc]: So method->options().GetExentsion() is always there even you did not specify the "options" in the proto file.  Is that right?

Comment (lizan) [inline src/envoy/transcoding/config.cc]: if you don't have options in the proto, GetExtension() will just return a default instance of HttpRule, which is fine.

**Pull request merged**